File: sdk/csharp/src/IO.StockX.Examples/SearchExample.cs
using System;
using System.Diagnostics;
using IO.StockX.Api;
using IO.StockX.Client;
using IO.StockX.Model;

namespace Example
{
    /**
    * <p>Demonstrates retrieving information StockX products from a general query.</p>
    */
    public class GetProductInfoExample
    {
        static public void Main()
        {
            // Configure API key authorization: api_key
            Configuration.Default.AddApiKey("x-api-key", ExampleConstants.AWS_API_KEY);

            var stockx = new StockXApi();
                stockx.Configuration.BasePath = ExampleConstants.STOCKX_ENDPOINT;
            
            var login = new LoginRequest(ExampleConstants.STOCKX_USERNAME, ExampleConstants.STOCKX_PASSWORD);
            
            try
            {
                // Login
                stockx.Login(login);
                
                // Lookup the supplied product by its product ID
                var search = stockx.Search("jeezy");

                Console.WriteLine(search);
            }
            catch (Exception e)
            {
                Console.WriteLine(e.ToString());
                Console.WriteLine("Exception when calling StockXApi.Search: " + e.Message );
            }
        }
    }
}

File: sdk/java/io.stockx.sdk.examples/src/main/java/io/stockx/sdk/examples/endpoints/SearchExample.java
package io.stockx.sdk.examples.endpoints;

import static io.stockx.sdk.examples.base.LoginUtil.login;

import io.stockx.client.ApiClient;
import io.stockx.client.ApiException;
import io.stockx.client.api.StockXApi;
import io.stockx.client.model.SearchResults;
import io.stockx.sdk.examples.base.Environment;
import io.stockx.sdk.examples.base.ExampleConstants;
import io.stockx.sdk.examples.base.StockXClient;

/**
 * <p>
 * Demonstrates retrieving information StockX products from a general query.
 * </p>
 */
public class SearchExample {

    public static void main(String[] args) {

        ApiClient apiClient = StockXClient.create(ExampleConstants.AWS_API_KEY, Environment.STAGING, true);

        StockXApi stockx = new StockXApi(apiClient);

        try {
            // Login to the StockX framework - see LoginExample for how this works
            login(stockx, ExampleConstants.STOCKX_USERNAME, ExampleConstants.STOCKX_PASSWORD);

            // Lookup the supplied product by its product ID
            SearchResults res = stockx.search("yeezy");

            System.out.println("Search result: " + res);
        } catch (ApiException e) {
            e.printStackTrace();
        }
    }

}
